File: app/src/main/java/com/nononsenseapps/feeder/ui/text/ClickableImageSpan.java
package com.nononsenseapps.feeder.ui.text;

import android.graphics.drawable.Drawable;
import android.text.style.ImageSpan;


/**
 * A clickable image span
 */
public abstract class ClickableImageSpan extends ImageSpan {
    public ClickableImageSpan(final Drawable d) {
        super(d);
    }

    public abstract void onClick();
}

File: app/src/main/java/com/nononsenseapps/feeder/ui/text/ClickableImageSpan.kt
package com.nononsenseapps.feeder.ui.text

import android.graphics.drawable.Drawable
import android.text.style.ImageSpan


/**
 * A clickable image span
 */
abstract class ClickableImageSpan(d: Drawable) : ImageSpan(d) {
    abstract fun onClick()
}
